monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js

```diff
@@ -175,6 +163,26 @@ class ConfigurePageComponent extends AuthComponent {
     fileDownload(JSON.stringify(this.state.configuration, null, 2), 'monkey.conf');
   };
 
+  sendConfig() {
+    return (
+      this.authFetch('/api/configuration/island',
+      {
+        method: 'POST',
+        headers: {'Content-Type': 'application/json'},
+        body: JSON.stringify(this.state.configuration)
+      })
+      .then(res => {
+        if (!res.ok)
```

Comment:
Why throw and catch immediately?

Reply:
I don't understand the question. You meant why this code doesn't catch it immediately?

Reply:
I mean why throw at all?

Reply:
How else would we know that invalid configuration was submitted? Either way, this is not my code, I just moved it into a function.